packages/safe-ethers-lib/src/EthersAdapter.ts

```diff
@@ -137,6 +138,16 @@ class EthersAdapter implements EthAdapter {
     return this.#signer.signMessage(messageArray)
   }
 
+  async signTypedData(safeTransactionEIP712Args: SafeTransactionEIP712Args): Promise<string> {
+    const typedData = generateTypedData(safeTransactionEIP712Args)
+    const signature = await (this.#signer as JsonRpcSigner)._signTypedData(
```

Comment:
Instead of casting, can't you initially type `#signer` as `JsonRpcSigner` instead of signer?

Reply:
Maybe you chould add a type guard to check that it supports the `_signTypedData` method?